src/VS.Mefx/Commands/Command.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace VS.Mefx.Commands
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.VisualStudio.Composition;
+
+    /// <summary>
+    /// A general command class which serves a parent class for all the commands that can be run by application .
+    /// </summary>
+    internal class Command
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Command"/> class.
+        /// </summary>
+        /// <param name="derivedInfo">The ConfigCreator for the input files.</param>
+        /// <param name="arguments">The command line arguments from the user.</param>
+        public Command(ConfigCreator derivedInfo, CLIOptions arguments)
+        {
+            this.Creator = derivedInfo;
+            this.Options = arguments;
+        }
+
+        /// <summary>
+        /// Write the given lines in sorted order to the user.
+        /// </summary>
+        /// <param name="lines">The lines to output to the user.</param>
+        protected void WriteLines(List<string> lines)
+        {
+            lines.Sort();
+            lines.ForEach(line => this.Options.Writer.WriteLine(line));
+        }
+
+        /// <summary>
+        /// Gets the composable catalog and configuration.
+        /// </summary>
+        protected ConfigCreator Creator { get; private set; } // Stores the catalog and config information for the input files.
+
+        /// <summary>
+        /// Gets the command line arguments passed in by the user.
+        /// </summary>
+        protected CLIOptions Options { get; private set; } // The command line arguments specified by the user.
+
+        /// <summary>
+        /// Method to get the name of the given its definition.
+        /// </summary>
+        /// <param name="part"> The defintion of the part whose name we want.</param>
+        /// <param name="verboseLabel"> Label to add before the verbose description of the part.</param>
+        /// <returns> A string representing either the simple or verbose name of the part based
+        ///          on if verbose was specified as an input argument.</returns>
```

Comment:
Please don't indent the next line of `<returns>`. For multiline xml elements, consider putting the tags on their own lines and the text in the lines in between.